server/functions/framework/compiled_function.go

```diff
@@ -88,17 +92,23 @@ func newCompiledFunctionInternal(
 	c.callResolved = make([]pgtypes.DoltgresType, len(functionParameterTypes)+1)
 	hasPolymorphicParam := false
 	for i, param := range functionParameterTypes {
-		if _, ok := param.(pgtypes.DoltgresPolymorphicType); ok {
+		// TODO: we use 'text' type for 'cstring' type, which is polymorphic type
+		if param.IsPolymorphicType() || param.OID == uint32(oid.T_text) {
 			// resolve will ensure that the parameter types are valid, so we can just assign them here
 			hasPolymorphicParam = true
 			c.callResolved[i] = originalTypes[i]
 		} else {
 			c.callResolved[i] = param
+			if d, ok := args[i].Type().(pgtypes.DoltgresType); ok {
+				// TODO: find better workaround to keep the type of the argument as parameter type
+				//  (they currently differ with type modifier information)
```

Comment:
The `param` type does not have the type modifier set, which is needed for function like `IoOutput` because the function does not pass in `typmod` like `IoInput` function.